src/server/tran_server.cpp

```diff
@@ -358,21 +360,30 @@ tran_server::uses_remote_storage () const
 }
 
 void
-tran_server::push_request (tran_to_page_request reqid, std::string &&payload)
+tran_server::push_request_to (size_t idx, tran_to_page_request reqid, std::string &&payload)
 {
+  assert (idx < m_page_server_conn_vec.size());
   if (!is_page_server_connected ())
```

Comment:
should be segregated in two:
- `are_all_page_server_connected`
- `is_page_server_connected(idx)`

and called according to function - here `is_page_server_connected(idx)`

Reply:
Good point, Thank you.

By the way, there is no use of `are_all_page_server_connected`. Who have used it usually checks if any ps is connected. So, we could just overload it like this:
- is_page_server_connected()      : check if at least one is available. (used as it is)
- is_page_server_connected(idx) : check if the specific one is available.

What do you think?

Reply:
>     is_page_server_connected() : check if at least one is available. (used as it is)
>     is_page_server_connected(idx) : check if the specific one is available.

Agreed.